### PR TITLE
KAN-43: Release blog robotics MongoDB vs ReductStore

### PR DESCRIPTION
**Pull request by AnthonyCvn**

### CODE REVIEW

File: blog/2024-11-26-robotics-mongodb/index.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+By understanding the strengths and limitations of each, you can make an informed decision to meet your project's specific data needs.
+
+{/* truncate */}
+
+**TL;DR;**
+
+| **Feature**                    | **ReductStore**                                                                           | **MongoDB**                                                                        |
+| ------------------------------ | ----------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------- |
+| **Architecture**               | Buckets, entries, records, and blocks                                                     | Collections of JSON-like documents                                                 |
```
Comment (atimin) [inline blog/2024-11-26-robotics-mongodb/index.mdx:25]: Here we compare ReductStore's internal and MongDB external architecture. Let's compare in general. 
- ReductStore: time series of string massages + meta information as a collection of key-values

File: blog/2024-11-26-robotics-mongodb/index.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+On the other hand, **MongoDB** uses its native _MongoDB Wire Protocol_, a lightweight, binary protocol optimized for fast communication between clients and servers. This protocol encodes data in BSON (Binary JSON), which supports efficient transmission and storage of JSON-like documents. These design choices make MongoDB a great choice for web-based applications, offering both flexibility and scalability.
+
+## Data Management Features
+
+### Data batching capabilities\*
+
+**ReductStore** enables efficient _data batching_ by grouping multiple records into a single HTTP request, minimizing delays and reducing the number of queries needed. It uses parameters like each_s (records per second) or each_n (specific record counts) to customize data retrieval, allowing applications to process only the required subsets efficiently. This batching mechanism supports use cases like real-time monitoring and historical data analysis with minimal network overhead.
```
Comment (atimin) [inline blog/2024-11-26-robotics-mongodb/index.mdx:61]: each_s and each_n is involved into querying but not batching. The storage batched data records into an HTTP request in an optimal way for the HTTP protocol

File: blog/2024-11-26-robotics-mongodb/index.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+**MongoDB**, provides _bulk_ _write operations_ that enable inserting, updating, or deleting multiple documents in a single transaction. This is particularly useful for applications that need to modify large datasets without compromising performance. MongoDB's aggregation pipelines also support batch data processing, but this batching is more general and not a good option for time-series data.
+
+### Edge-to-Cloud Replication
+
+**MongoDB's** replication system mainly focuses on maintaining data consistency and fault tolerance across multiple replica sets. While it is very reliable and flexible for general-purpose applications, it is not specifically suited for edge-to-cloud workflows since it does not natively allow data filtering during the replication process.
```
Comment (atimin) [inline blog/2024-11-26-robotics-mongodb/index.mdx:67]: order: in other sections ReductStore comes first.

File: blog/2024-11-26-robotics-mongodb/index.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+By understanding the strengths and limitations of each, you can make an informed decision to meet your project's specific data needs.
+
+{/* truncate */}
+
+**TL;DR;**
+
+| **Feature**                    | **ReductStore**                                                                           | **MongoDB**                                                                        |
+| ------------------------------ | ----------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------- |
+| **Architecture**               | Buckets, entries, records, and blocks                                                     | Collections of JSON-like documents                                                 |
+| **Query Language**             | Query data via CLI, SDKs, or HTTP API                                                     | MongoDB Query Language (MQL)                                                       |
```
Comment (atimin) [inline blog/2024-11-26-robotics-mongodb/index.mdx:26]: in the current version, we don't have any query language. I would say something like:

Data Queering:
* ReductStore: query requests via time ranges with filters
* MongoDB Query Language (MQL)

**Review by atimin — CHANGES_REQUESTED** [review]
In general it's good but there are some technical details.  I'd also suggest to add  few sentences in each feature section with explanation why it's important for robots.  

For example,

## Data batching capabilities

Many autonomous robotic system work in a networks with high latency that's why the ability to batch or stream data is a critical for data availability..

**Comment by atimin** [comment]
> In general it's good but there are some technical details. I'd also suggest to add few sentences in each feature section with explanation why it's important for robots.
> 
> For example,
> 
> ## Data batching capabilities
> Many autonomous robotic system work in a networks with high latency that's why the ability to batch or stream data is a critical for data availability..

Hey @gracija , nice to meet you =). Could it be possible to add a few short sentences in the feature sections to emphasize why they are important for robotic systems?

**Comment by AnthonyCvn** [comment]
@atimin she's on a trip; but I could add a few sentences.

**Review by atimin — APPROVED** [review]
Perfect!

**Pull request merged**